### PR TITLE
External dependencies upgrade - Spring Boot 3.0.0-RC2 and Spring Cloud 2022.0.0-RC2

### PR DESCRIPTION
**Pull request by moarychan**
Upgrade the dependencies to align with Spring Boot 3.0.0-RC2 and Spring Cloud 2022.0.0-RC2.

### CODE REVIEW

File: eng/versioning/external_dependencies.txt [hunk 1]
```diff
@@ -66,15 +66,15 @@ io.netty:netty-tcnative-boringssl-static;2.0.54.Final
...
-io.projectreactor.netty:reactor-netty-http;1.1.0-RC1
-io.projectreactor:reactor-core;3.5.0-RC1
+io.projectreactor.netty:reactor-netty;1.1.0
+io.projectreactor.netty:reactor-netty-http;1.1.0
+io.projectreactor:reactor-core;3.5.0
 io.reactivex:rxjava;1.3.8
 io.vertx:vertx-codegen;4.3.3
 io.vertx:vertx-core;4.3.3
-jakarta.servlet:jakarta.servlet-api;5.0.0
+jakarta.servlet:jakarta.servlet-api;6.0.0
```
Comment (saragluna) [inline eng/versioning/external_dependencies.txt:75]: Do you know why this is a major version upgrade?

File: eng/versioning/external_dependencies.txt [hunk 2]
```diff
@@ -66,15 +66,15 @@ io.netty:netty-tcnative-boringssl-static;2.0.54.Final
...
 jakarta.validation:jakarta.validation-api;3.0.2
-jakarta.xml.bind:jakarta.xml.bind-api;3.0.1
+jakarta.xml.bind:jakarta.xml.bind-api;4.0.0
```
Comment (saragluna) [inline eng/versioning/external_dependencies.txt:77]: Same for this one.

File: eng/versioning/external_dependencies.txt [hunk 3]
```diff
@@ -102,7 +102,7 @@ org.codehaus.groovy:groovy-jsr223;3.0.13
 org.codehaus.janino:janino;3.1.8
 org.conscrypt:conscrypt-openjdk-uber;2.2.1
 org.glassfish.tyrus:tyrus-client;1.16
-org.hibernate.validator:hibernate-validator;7.0.5.Final
+org.hibernate.validator:hibernate-validator;8.0.0.Final
```
Comment (saragluna) [inline eng/versioning/external_dependencies.txt:105]: Same

File: eng/versioning/external_dependencies.txt (hunk 1 above)
Comment (moarychan) [reply eng/versioning/external_dependencies.txt:75]: Upgrade to Jakarta EE 10, https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.0.0-RC2-Release-Notes#jakarta-ee-10

**Review by saragluna — APPROVED** [review]

File: .vscode/cspell.json [hunk 4]
```diff
@@ -208,10 +208,7 @@
     "sdk/spring/spring-cloud-azure-starter-stream-servicebus/**",
     "sdk/spring/spring-cloud-azure-autoconfigure/**",
     "sdk/spring/spring-cloud-azure-stream-binder-eventhubs/**",
-    "sdk/spring/spring-cloud-azure-test-servicebus-binder/**",
-    "sdk/spring/spring-cloud-azure-trace-sleuth/**",
     "sdk/spring/spring-cloud-azure-stream-binder-eventhubs-core/**",
-    "sdk/spring/spring-cloud-azure-test-eventhubs-binder/**",
```
[the remark this answers is not in the document]
Comment (moarychan) [reply .vscode/cspell.json:214]: https://github.com/Azure/azure-sdk-for-java/pull/32272

**Pull request merged**